src/snowflake/connector/chunk_downloader.py

```diff
@@ -315,7 +272,7 @@ def get_timings(self):
         return self._timings
 
 
-class RawBinaryDataHandler:
+class RawBinaryDataHandler(ABC):
     """Abstract class being passed to network.py to handle raw binary data."""
 
     def to_iterator(self, raw_data_fd, download_time):
```

Comment:
Do you mean to use '@abstractmethod' for to_iterator with introducing 'ABC'?

Reply:
> LGTM. Still some redundant code can be shared between Json and Arrow. Also the logic in chunk downloader is to prefetch ahead and this can be done in this result_set/result_batch implementation. We should monitor the performance difference related to this logic.

I'm sorry I'm working through your comments one-by-one and that's the last thing that I left. I agree with you and am working on reducing the redundant code. Sorry, I maybe should have been more upfront with you and told you not to check the PR every time I push. I often do that to verify that I haven't broken any tests.

Reply:
> Do you mean to use '@AbstractMethod' for to_iterator with introducing 'ABC'?

I'm sorry, I made this change in the very early days of me working on this. I'll roll it back. The while `chunk_downloader.py` is getting deleted.